src/components/pages/Home/Home.tsx

```diff
@@ -1,19 +1,62 @@
 import SearchBar from '../../inputs/Search/SearchBar/SearchBar';
 import './styles.css';
+import { useStateValue } from '../../../state/state';
+import { useState, useEffect } from 'react';
+import AvailabilityHeader from '../../layout/AvailabilityHeader/AvailabilityHeader';
 
 function Home() {
+  const [{ arnsSourceContract }] = useStateValue();
+  const [records, setRecords] = useState({});
+
+  const [availability, setAvailability] = useState('search');
+  const [searchName, setSearchName] = useState('');
+
+  const [placeholderText, setPlaceholderText] = useState('Find a name');
+
+  useEffect(() => {
+    let newRecords = arnsSourceContract.records;
+    setRecords(newRecords);
+  }, [arnsSourceContract]);
+
+  function availiblityCheck(props: string) {
+    // if name is not in the legal character range or chars, dnt run
+    const namePattern = new RegExp('^[a-zA-Z0-9_-]{1,32}$');
+    if (!namePattern.test(props) && props !== '') {
+      console.log('not a valid name', namePattern.test(props));
+      return;
+    } else {
+      //if empty reset to search state
+      if (props == '') {
+        setAvailability('search');
+        setPlaceholderText('Find a name');
+      }
+      //if not registered set as available
+      if (!Object.keys(records).includes(props) && props !== '') {
+        setAvailability('available');
+        setSearchName(props);
+      }
+      //if registered set as unavailable
+      if (Object.keys(records).includes(props) && props !== '') {
+        setAvailability('unavailable');
+        setSearchName(props);
+        setPlaceholderText('try another name');
+      }
+    }
+  }
+
   return (
     <div className="page">
       <div className="pageHeader">Arweave Name System</div>
       <SearchBar
-        buttonAction={() => {
-          console.log('search');
-        }}
-        placeholderText="Enter a name"
-        headerText={'Find a domain name'}
+        buttonAction={(props) => availiblityCheck(props)}
+        placeholderText={placeholderText}
+        headerElement={
+          <AvailabilityHeader availability={availability} name={searchName} />
+        }
         footerText={
           'Names must be 1-32 characters. Dashes and underscores are permitted, but cannot be trailing characters and cannot be used in single character domains.'
         }
+        availability={availability}
```

Comment:
nit: `isAvailable` is bit more more clear and should a `bool`

Reply:
bool is binary and doesnt cover the initial search state. availability has search, available, and unavailable as rendering conditions do to the searchbar having the those three states.

Reply:
so our states are 
```
if text is empty - show default state
if text is not empty:
    - name is valid and available - success state
    - name is INVALIDA or NOT AVAILABILE - error state
```